src/net/socket.cpp

```diff
@@ -103,18 +101,38 @@ Socket::port_type Socket::bind() {
 }
 
 IPv4 Socket::address() const {
+	if (not Init::initialized()) {
```

Comment:
I don't like the fact that this code is duplicated all over the place. IMO we don't need to check whether system is initialized all the time, but if you really want to do this, then maybe at least think if there is a way to avoid duplication and maybe don't do that for functions that don't need initialization (in this case you just return _address;